### DISCUSSION TITLE
Forcing xbps-src to resolve dependencies from binpkg repos

### DISCUSSION THREAD

**Discussion by gmbeard** [void-linux/void-packages#53567] [Q&A] [answered]
What would be the recommended way to force `./xbps-src` to resolve its dependencies from the binary pkg repo instead of srcpkgs?

For context... I have a python3 utility that I want to pkg and install locally. At the time of writing, the version of python in srcpkgs is `3.13.1` but the version available in the binary pkg repo is `3.12.7`. Therefore, if I build my pkg my utility using its srcpkg, it uses `python3-3.13.1` as its dependency, and I get a package that I cannot install/use locally (it depends on a module that wants to install itself in the wrong version of `site-packages`).

As far as I'm aware, there are two ways to do this...

1. Temporarily revert the python3 commit to srcpkgs to roll it back to the repo version
2. Hack [some questionable changes][1] into `./xbps-src` to force the use of the repo version

I fully understand that this is probably quite a niche scenario, but I would grateful for any help in understanding how this works.

[1]: https://gist.github.com/gmbeard/3c58c41b379fb71c6e899da195ea454f

**Comment by classabbyamp** [comment, marked as answer]
if you build the offending package(s) once on your master branch, other branches can use them. otherwise, revert, checkout a commit before the push, or wait.
